### ISSUE TITLE
`Button` component should render `<button>` when no `url` is defined

### ISSUE THREAD

**Issue by th3hamm0r** [wagtail/wagtail#13250]
### Is your proposal related to a problem?

We are using Wagtail as an "admin framework" quite intensively atm and we often have the case, where we need a `<button>` in the header buttons or the list buttons, which triggers a POST-request or opens a dialog.
For that we've created a custom `Button` base class, which uses the following template:
```html
{% load wagtailadmin_tags %}
{% if button.url %}
    {% include "wagtailadmin/shared/button.html" %}
{% else %}
    <button class="{{ button.classname }}" {{ button.base_attrs_string }}>
        {% if button.icon_name %}
            {% icon name=button.icon_name wrapped=1 %}
        {% endif %}
        {{ button.label }}
    </button>
{% endif %}
```
The Button class is simple:
```python
from wagtail.admin.widgets.button import Button as BaseButton

class Button(BaseButton):
    template_name = "utils/wagtailadmin/shared/button.html"
```

We then have subclasses like the following, which is used to render such a button in a listing table:
```python
from wagtail.admin.widgets import ListingButton

class ActionListingButton(Button, ListingButton):
    def __init__(self, label="", url=None, classname="", **kwargs):
        attrs = kwargs.setdefault("attrs", {})
        attrs["data-controller"] = (
            f"{attrs.get('data-controller', '')} w-progress w-action".strip()
        )
        attrs["data-action"] = "w-progress#activate w-action#post"
        attrs["data-w-action-url-value"] = url
        super().__init__(label, url=None, classname=classname, **kwargs)
```

This works quite well for us, so I was thinking if we can get this template logic into Wagtail's core [button.html](https://github.com/wagtail/wagtail/blob/e75bbca5e2d8acd8c7bc434856547b3da4526bd5/wagtail/admin/templates/wagtailadmin/shared/button.html)?
(I know, for us it was a simple change, but since we don't use every feature and module of Wagtail, I don't have the full overview if this would have some negative effects 😅 )

### Describe the solution you'd like

The [template](https://github.com/wagtail/wagtail/blob/e75bbca5e2d8acd8c7bc434856547b3da4526bd5/wagtail/admin/templates/wagtailadmin/shared/button.html) could be changed to the following:
```html
{% load wagtailadmin_tags %}
{% if button.url %}
    <a href="{{ button.url }}" class="{{ button.classname }}" {{ button.base_attrs_string }}>
        {% if button.icon_name %}
            {% icon name=button.icon_name %}
        {% endif %}
        {{ button.label }}
    </a>
{% else %}
    <button class="{{ button.classname }}" {{ button.base_attrs_string }}>
        {% if button.icon_name %}
            {% icon name=button.icon_name wrapped=1 %}
        {% endif %}
        {{ button.label }}
    </button>
{% endif %}
```

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by nayelimdejesus** [comment]
I will be working on this issue.

**Comment by tomkins** [comment]
It would be nice if a button could be:

- A link (as it is currently).
- A button that can be tweaked/customised with javascript (just type=button)
- A button inside a small `<form>` that submits to a URL (either the current page, or a custom URL).

But the underlying big dream I'd love to see in Wagtail - for any ModelViewSet (and perhaps other things in Wagtail), it would be great if you could easily add actions in the admin for any type of object without having to write view code or template code. Something similar to BulkAction - which is pretty good, but doesn't support ModelViewSet (see #11371), but does require a template. And something similar to Django admin's bulk actions - that allows the developer to decide if they need an intermediate page or not (and is super easy to implement if a page isn't needed).

**Comment by th3hamm0r** [comment]
I'm not sure, this sounds like a bit too much logic for this basic button component 🤔 
I even hesitated creating the above issue, but I think this basic switch between link and button gives a lot of opportunities without overriding the template (just by passing kwargs like `attrs`).

Nowadays the "Admin UI API" of Wagtail is quite easy and straight forward. Just subclass the component (like Buttons, Panels,...) or view class, and override some methods or create new templates.
With hooks you always have the problem that they only allow to tweak one very small portion of the functionality, and the Wagtail core just can't implement hooks for every feature one even doesn't think of 😅 
Of course, there are plenty of things that could be improved, but I really like the current state, and the path the API is going, as you are able to add components to various places in the UI very quickly with only a view lines of code.

**Comment by PrathmeshDesai** [comment]
Hello , is these issue is fixed ?

**Comment by laymonage** [comment]
Sounds like a reasonable request, thanks!

> * A button inside a small `<form>` that submits to a URL (either the current page, or a custom URL).

I don't think this should be incorporated into the `Button` component. This can be easily implemented as its own component that wraps a `Button` (that has `type="submit"`) in a `<form>` element.
